src/main.rs

```diff
@@ -222,7 +220,25 @@ async fn handle_request(
         }
     };
 
-    let trimmed_path = request_path.strip_prefix(&api_url).unwrap_or(&request_path);
+    let (api_path, trimmed_path) = if let Some(trimmed_path) = request_path.strip_prefix("/api") {
+        if let Some(maybe_api_version) = trimmed_path.split('/').nth(1) {
+            if let Some(version_number) = maybe_api_version.strip_prefix("v") {
+                if version_number.parse::<u8>().is_ok() {
+                    // 6 = /api/v
+                    let len = 6 + version_number.len();
+                    (&request_path[..len], &request_path[len..])
+                } else {
+                    ("/api", trimmed_path)
+                }
+            } else {
+                ("/api", trimmed_path)
+            }
+        } else {
+            ("/api", trimmed_path)
+        }
```

Comment:
Can't this be 
```suggestion
                }
            }
        } else {
            ("/api", trimmed_path)
        }
```

Reply:
not really from what the compiler says:
```rs
error[E0317]: `if` may be missing an `else` clause
   --> src/main.rs:226:17
    |
226 | /                 if version_number.parse::<u8>().is_ok() {
227 | |                     // 6 = /api/v
228 | |                     let len = 6 + version_number.len();
229 | |                     (&request_path[..len], &request_path[len..])
    | |                     -------------------------------------------- found here
230 | |                 }
    | |_________________^ expected `()`, found tuple
    |
    = note: expected unit type `()`
                    found type `(&str, &str)`
    = note: `if` expressions without `else` evaluate to `()`
    = help: consider adding an `else` block that evaluates to the expected type
```

Reply:
moving it into a function should work though, i'll do that